### PR TITLE
Update SchemaFields to include field types.

### PR DESCRIPTION
**Pull request by csilvestrini**
Field types are needed so that we can map entities via their schemas into the normalised database structure.

### CODE REVIEW

**Comment by jasonwyatt** [comment]
I think we may need a way to pass an entity schema down through the storage proxy, store, and driver. Currently none of that info is available so close to the drivers

File: java/arcs/core/data/SchemaFields.kt [hunk 1]
```diff
@@ -11,8 +11,40 @@
...
+sealed class FieldType(
+    val tag: Tag
+) {
+    /** An Arcs primitive type. */
+    data class Primitive(val primitiveType: PrimitiveType): FieldType(Tag.Primitive)
+
+    /** A reference to an entity. */
+    data class EntityRef(val schemaHash: String) : FieldType(Tag.EntityRef)
+
+    // TODO: Collections of References in fields are not supported in Kotlin at all yet.
```
Comment (jasonwyatt) [inline java/arcs/core/data/SchemaFields.kt]: They aren't? You should be able to make a collection of anything referencable, including a dumb reference instance which only has a storage key.

**Comment by csilvestrini** [comment]
> I think we may need a way to pass an entity schema down through the storage proxy, store, and driver. Currently none of that info is available so close to the drivers

I think you're right. We need to have knowledge of the schema at the level where we interact with the database, so it's going to have to be plumbed the whole way..

File: java/arcs/core/data/SchemaFields.kt (hunk 1 above)
Comment (csilvestrini) [reply java/arcs/core/data/SchemaFields.kt]: I was thinking more at the public sdk level. You're right though, the storage stack should support it. I think I can just delete this comment, it's actually not needed here. The knowledge of whether the field is a singleton or collection is kept elsewhere (in the SchemaFields singleton and collection fields).

**Comment by csilvestrini** [comment]
cl/291096133 passing

**Pull request merged**